example/http/client/sync/CMakeLists.txt

```diff
@@ -41,6 +41,7 @@ add_executable (http-client-sync-post-json
 )
 
 target_link_libraries(http-client-sync-post-json
+    boost_json
```

Comment:
Boost::json should work.

Reply:
@pdimov I doesn't for me:

```
|| CMake Error at example/json-body/CMakeLists.txt:19 (target_link_libraries):
||   Target "http-client-json-body" links to:
|| 
||     Boost::json
|| 
||   but the target was not found.  Possible reasons include:
|| 
||     * There is a typo in the target name.
||     * A find_package call is missing for an IMPORTED target.
||     * An ALIAS target is missing.
```

Reply:
@pdimov Seems I can force the issue by slipping in `json` in the required components in a seemingly random place in the Beast global CMakeLists.txt: https://github.com/boostorg/beast/blob/develop/CMakeLists.txt#L192

However, we may not want that because Boost JSON should not become a Beast dependency. In fact, that's exactly why `json_body` is only gonna be an example.

Reply:
Yeah, I was just going to suggest that. The place must not be so random.

Reply:
I don't understand why it is in a conditional section only "unless MSVC is involved"(?). That's why I'd normally never stuff it in there. But it was the only `find_package(Boost` in sight, so I tried it. Do you reckon we can get away with that or should we just keep it as is due to dependency management?

Reply:
Linking to `boost_json` just adds `-lboost_json` to the link command. Whether the library will then be found is anyone's guess. A "normal" project needs to find Boost.JSON in order to link to it.

I suppose you can add the `find_package(Boost REQUIRED COMPONENTS json)` to the `json-body` CML file, but I'm not sure whether FindBoost supports multiple calls like that.

Reply:
That appears to work, I see that the sub-directory reports the superset of required components, so at least something here seems to "interfere fortunately" - I'm assuming that might just mean it works. We'll see whether CI barfs on it I guess when I push it tomorrow to test.